server/polar/account/service.py

```diff
@@ -55,53 +86,89 @@ async def create_account(
             data=stripe_account.to_dict(),
         )
 
-    async def onboarding_link_for_user(
+    async def _create_open_collective_account(
         self,
         session: AsyncSession,
         organization_id: UUID,
-        user: User,
-        stripe_id: str,
-        appendix: str | None = None,
-    ) -> AccountLink | None:
-        account = await self.get_by(
-            session=session, organization_id=organization_id, stripe_id=stripe_id
-        )
-        if account is None or account.admin_id != user.id:
-            # TODO: Error?
-            return None
+        admin_id: UUID,
+        account: AccountCreate,
+    ) -> Account:
+        assert account.open_collective_slug is not None
+        try:
+            collective = await open_collective.get_collective(
```

Comment:
We have a service to query the Open Collective API and check if the collective exists, is valid and has the right financial host